### DISCUSSION TITLE
Zod abi parser fails when the name of input or output is an empty string.

### DISCUSSION THREAD

**Discussion by StanleyDing** [wagmi-dev/abitype#184] [Q&A]
Given this example abi fetched from etherscan, zod parser produces error saying the `name` field in the ouput is invalid when it's an empty string.

```ts
import { Abi as ZodAbi } from "abitype/zod";

const abi = [
  {
    "inputs": [
      {
        "internalType": "address",
        "name": "account",
        "type": "address"
      }
    ],
    "name": "balanceOf",
    "outputs": [
      {
        "internalType": "uint256",
        "name": "",  // <---------- Empty!
        "type": "uint256"
      }
    ],
    "stateMutability": "view",
    "type": "function"
  }
];

ZodAbi.parse(abi)
```

Error:
```text
Uncaught ZodError: [
  {
    "validation": "regex",
    "code": "invalid_string",
    "message": "Invalid",
    "path": [
      0,
      "outputs",
      0,
      "name"
    ]
  }
]
```

I don't see empty sring is not allowed in the [abi spec](https://docs.soliditylang.org/en/latest/abi-spec.html#json), so perhaps we should allow empty string as valid name of function parameter?

I could send a PR to fix this if devs agree.

**Comment by Raiden1411** [comment]
Empty string on unnamed parameters are the default when Solidity compiles a contract so this should be supported. Feel free to make a PR to fix it

**Comment by StanleyDing** [comment]
Submitted a PR in #185.
